### ISSUE TITLE
Switch to assertEqual and the other TestCase assertion methods which are not deprecated in the standard library

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#4990]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#4990|
|Type|enhancement|
|Created|2011-03-26 19:36:12Z|
|Branch|https://github.com/twisted/twisted/tree/use-assertEqual-4990|

Per the mailing list discussion, [TestCase methods names](http://twistedmatrix.com/pipermail/twisted-python/2011-March/023681.html), Twisted's own test suite should not use the assertion methods which are deprecated in the standard library.

<details><summary>Searchable metadata</summary>

```
trac-id__4990 4990
type__enhancement enhancement
reporter__exarkun exarkun
priority__normal normal
milestone__ 
branch__branches_use_assertEqual_4990 branches/use-assertEqual-4990
branch_author__cyli cyli
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1301168172000000 1301168172000000
changetime__1310670327000000 1310670327000000
version__None None
owner__cyli cyli

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_1"></a>|@cyli commented|
|-|-|

(In [32217]) Branching to 'use-assertEqual-4990'

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_2"></a>|@cyli commented|
|-|-|

Either the change has not been propagated to the site yet, or I messed up my refs tag: [32218](http://twistedmatrix.com/trac/changeset/32218)

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_3"></a>|@cyli commented|
|-|-|

[build results](http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/use-assertEqual-4990)

Almost everything was replaced with a `"grep -Irl … | xargs sed"` command to replace `self.failUnlessEquals`, `self.failUnlessEqual`, and `self.assertEquals` with `self.assertEqual` (such occurrences were only in the test directories), except for the following:

`twisted/lore/test/test_lore.py` - defines several variations assertEqualsFiles methods, so file was edited manually

`twisted/news/test/test_nntp.py` - defines its own assertEquals method, unchanged

After the automated changes, I also manually went through and changed occurrences of "failUnlessEqual" and "assertEquals" as needed.

Also:

`twisted/trial/test/test_assertions.py` - test_failUnlessEqual* names and error messages and comments were changed to test_assertEqual*

`twisted/trial/unittest.py` - changed the actual defined method name from `failUnlessEqual` to `assertEqual`

**Comment by twisted-trac** [comment]
|[<img alt="dustin's avatar" src="https://avatars.githubusercontent.com/u/28673?s=50" width="50" height="50">](https://github.com/dustin)<a name="note_4"></a>|@dustin set owner to @cyli|
|-|-|

Whew, big patch - thanks for the guide to the interesting parts!

In ```test_nntp.py```, I don't see a good reason to keep the incorrectly-spelled method name.  Why not ```s/assertEquals/assertEqual/g``` on that file, too?

I'm happy with it other than that detail.

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_5"></a>|@cyli commented|
|-|-|

(In [32337]) Change test_nntp's assertEquals definition to assertEqual instead

refs [#4990](https://github.com/twisted/twisted/issues/4990)

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_6"></a>|@cyli commented|
|-|-|

Thanks for the review Dustin!  I'm interpreting "I'm happy with it other than that detail" to mean "please merge after dealing with my concerns".  :)

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_7"></a>|@cyli set status to `closed`|
|-|-|

(In [32338]) Merge use-assertEqual-4990: change failUnlessEqual(s) and assertEquals to assertEqual

Author: cyli
Reviewer: dustin
Fixes: [#4990](https://github.com/twisted/twisted/issues/4990)

Change usage of failUnlessEqual(s) and assertEquals in twisted code to assertEqual
